src/service/folder_service.rs

```diff
@@ -753,3 +753,125 @@ mod download_folder_tests {
+        
+        cleanup();
+    }
+
+    #[test]
+    fn update_folder_propagates_tags_to_descendant_files() {
+        init_db_folder();
+        // Create folder hierarchy with files
+        create_folder_disk("top/middle");
+        create_folder_db_entry("top", None); // 1
+        create_folder_db_entry("middle", Some(1)); // 2
+        create_file_db_entry("file1", Some(1)); // file in top
+        create_file_db_entry("file2", Some(2)); // file in middle
+        
+        // Update top folder with a tag
+        let update_req = UpdateFolderRequest {
+            id: 1,
+            name: "top".to_string(),
+            parent_id: None,
+            tags: vec![TagApi { id: None, title: "tag1".to_string() }],
+        };
+        update_folder(&update_req).unwrap();
+        
+        // Check that both files inherited the tag
+        let con = open_connection();
+        let file1_tags = tag_repository::get_tags_on_file(1, &con).unwrap();
+        let file2_tags = tag_repository::get_tags_on_file(2, &con).unwrap();
+        con.close().unwrap();
+        
+        assert_eq!(1, file1_tags.len());
+        assert_eq!("tag1", file1_tags[0].title);
+        assert_eq!(1, file2_tags.len());
+        assert_eq!("tag1", file2_tags[0].title);
+        
+        cleanup();
+    }
+
+    #[test]
+    fn update_folder_removes_tags_not_on_parent() {
+        init_db_folder();
+        // Create folder hierarchy
+        create_folder_disk("top/child");
+        create_folder_db_entry("top", None); // 1
+        create_folder_db_entry("child", Some(1)); // 2
+        create_file_db_entry("file", Some(2)); // file in child
+        
+        // Update top folder with tag1
+        let update_req = UpdateFolderRequest {
+            id: 1,
+            name: "top".to_string(),
+            parent_id: None,
+            tags: vec![TagApi { id: None, title: "tag1".to_string() }],
+        };
+        update_folder(&update_req).unwrap();
+        
+        // Verify child has tag1
+        let con = open_connection();
+        let child_tags = tag_repository::get_tags_on_folder(2, &con).unwrap();
+        con.close().unwrap();
+        assert_eq!(1, child_tags.len());
+        
+        // Now update top folder to remove all tags
+        let update_req2 = UpdateFolderRequest {
+            id: 1,
+            name: "top".to_string(),
+            parent_id: None,
+            tags: vec![],
+        };
+        update_folder(&update_req2).unwrap();
+        
+        // Verify child no longer has tag1 (it was inherited and parent no longer has it)
+        let con2 = open_connection();
+        let child_tags2 = tag_repository::get_tags_on_folder(2, &con2).unwrap();
+        let file_tags = tag_repository::get_tags_on_file(1, &con2).unwrap();
+        con2.close().unwrap();
+        
+        assert_eq!(0, child_tags2.len());
+        assert_eq!(0, file_tags.len());
+        
+        cleanup();
+    }
+}
```

Comment:
Also add tests to make sure that direct tags are unchanged for both folders and files when an ancestor changes its tags